codestyle/errorprone-rules.properties

```diff
@@ -227,6 +227,12 @@ UseCorrectAssertInTests=ERROR
 ConstantPatternCompile=ERROR
 # Variables initialized with Pattern#compile calls on constants can be constants
 
+ObjectsHashCodePrimitive=ERROR
+# Objects.hashCode(Object o) should not be passed a primitive value
+
+OptionalMapToOptional=ERROR
+# Mapping to another Optional will yield a nested Optional. Did you mean flatMap?
```

Comment:
This is the only one I'm not so sure about. The bug you found looks real to me, but this is a legit pattern and I'm a bit wary of disallowing it

Reply:
The need to nested `Optional` is quite hypothetical, but the error pattern is common in my experience. I think the check is worth enforcing for now. We can remove it if/when we have to.

Reply:
I agree with @dimas-b; and in the unlikely case where there is a compelling reason to have `Optional<Optional<...>>`, we can also add a `@SuppressWarnings` annotation.